### ISSUE TITLE
Part/PartDesign: Request for ability to trim surfaces

### ISSUE THREAD

**Issue by FreeCAD-Bug-Importer** [FreeCAD/FreeCAD#5587]
## Feature description
The ability to trim a surface based on it's intersection with another surface would be a time-saver when modelling.  For example in the attached screen grab it would be handy to be able to trim off the upper part of the vertical surface above the horizontal surface in a point and click manner.

From the following forum thread:
http://forum.freecadweb.org/viewtopic.php?f=3&t=6706

## Other information

Issue imported from https://tracker.freecad.org/view.php?id=1567 by **bejant** on 5/30/2014

**Comment by luzpaz** [comment]
@donovaly does this ticket have anything to do with the work you've been doing on #8252 ?

**Comment by donovaly** [comment]
> @donovaly does this ticket have anything to do with the work you've been doing on #8252 ?

No.

However this bug report can be closed. It does not provide sufficient info what exactly is missing. Since 2014 we have different tools to cut surfaces.

**Comment by luzpaz** [comment]
An aside, anyone know bejant's github handle?

**Comment by donovaly** [comment]
> @donovaly there is a forum thread with more details, did you check that before closing ticket?

I checked the linked forum thread. Last entry from 2014, no clear description what is missing and we have now the surface WB. We also have improved cut tools in Part WB.

If the problem persists, a new ticket with more details can be created.

In general, I tend to close issues when it is not 100 % clear what the case is. If there is still an issue, it is better to get a new one. A ticket that has not been touched for years will not be on the radar. Also it cost a lost of time for the devs to check what an old ticket might be about, if still valid etc. I therefore like the bug janitor in Bugzilla that closes such bugs automatically after some time. It issues you a warning that it will be closed, once not re-confirmed. If nobody reacts it is closed. I think we need such feature also for us.

**Comment by luzpaz** [comment]
The point of migrating to github was to also review open tickets. And there hasn't been a systematic review of open tickets by the devs (only during releases and we've had just one release since). Beside me there have been only a few folk that have taken deep dives in to the issue queue.  IMHO, just closing tickets because their old doesn't make sense. It's like "throwing the baby out with the bath water".

**Comment by wwmayer** [comment]
I disagree. The request is quite clear to me and currently not or not easily doable in Part or Surface wb.

Consider two intersecting faces or shells A and B where you want to trim face/shell A with face/shell B. The function is supposed to allow the user to click the part of the face/shell of A that should be kept.

**Comment by donovaly** [comment]
> Beside me there have been only a few folk that have taken deep dives in to the issue queue. IMHO, just closing tickets because their old doesn't make sense. It's like "throwing the baby out with the bath water".

Oh, I appreciate your outstanding work!
It is that we have so many open feature requests that nobody looks at if they are still valid. Also note that on other projects, it is common to close feature requests as "WONTFIX", meaning nobody has time, too complicated etc. I have been part in different OpenSource development projects and in my opinion out of my experience the janitor service that closes feature requests after some time, is the best I found so far.
Note that this applies only for feature requests, not bugs. Verified bugs must stay of course open.
Also, this said, if  of course by subjective opinion.

Regarding this issue, i looked at the forum thread and to me it was not clear what exactly is missing that could not  be done using existing tools. (my approach in mind was to either bad to the desired surface then cut or use the surface WB to match a free-form surface.

**Comment by donovaly** [comment]
> Consider two intersecting faces or shells A and B where you want to trim face/shell A with face/shell B. The function is supposed to allow the user to click the part of the face/shell of A that should be kept.

OK.
However, for me it is not clear how this feature could be implemented. I mean using only the Part WB allows you to cut the surface, here what I mean:
[test.zip](https://github.com/FreeCAD/FreeCAD/files/10701267/test.zip)

Maybe I misunderstood the report.

**Comment by PLChris** [comment]
Sketcher has a trim tool which removes a part of a geometric element that is crossed by some other geometric element.
If we replace "geometric element" by "surface" we get what such a tool should do. In 3D things are a bit more complicated, because we need as a side condition that the surface to be trimmed is indeed completele traversed by the other surface.

Reusing donovaly's example, I think this should happen: Clicking on the green part of the vertical surface ...
![SnipScreenshot-409841](https://user-images.githubusercontent.com/11750017/218019559-d7986e43-5aad-4f16-817a-310c5e34e997.png)
... removes it and leaves you with this:
![SnipScreenshot-00f28d](https://user-images.githubusercontent.com/11750017/218019687-d28114e7-b4d4-4689-8a20-782f715aeca7.png)
It could well be implemented on top of slice apart, which I used for the example shown here.

**Comment by PLChris** [comment]
Moving discussions to Github was said to have the advantage of not having linear discussions only. How can that be done with the off topic "what to close" discussion here?

**Comment by maxwxyz** [comment]
Demo of split with surface in Part Design on Link branch: https://youtu.be/qPoFrgk8S2U?si=uDGPZ87rcSlP_Lyr&t=1320

**Comment by maxwxyz** [comment]
Is there any news on this issue? It is implemented in the Link branch @realthunder maybe you can weigh in.
@wwmayer you are currently assigned to this issue, is this still valid or do you want me to unassign you?

**Comment by lucas-bueno777** [comment]
Hello! I could add that the ability to trim surfaces is a daily tool when dealing with complex shapex. From the CATIA generative surface workbench, there are 2 tools that can cut a surface: "Split" and "Trim". The behavior It is a bit different for each of them: Split allows the user to use cutting elements to effectively cut the shape, for example you can use two planes to cut a rectangle and you can also decide which one from the 3 pieces to keep, potentially keeping all of them if you check an option. Therefore, you can have up to multiple resulting surfaces that can be used independently afterwards.

Meanwhile, "Trim" command allows multiple intersecting surfaces to "glue" to each other and for each new surface one selects to trim, you can choose which side will be kept and which one is thrown away. The resulting manifold geometry is a single surface composed of multiple trimmed surfaces. This is a very powerful tool!

https://www.youtube.com/watch?v=4E1S_SXejxI&ab_channel=CADTutorialsForBeginnersToAdvanced

**Comment by GAZ082** [comment]
This feature request is dead, looks like nobody is working on this. Also, looks like its parent is not in the 26.3 milestone.
